contracts/validator-manager/tests/ERC20TokenStakingManagerTests.t.sol

```diff
@@ -38,29 +33,31 @@ contract ERC20TokenStakingManagerTest is PoSValidatorManagerTest {
         app.initializeValidatorSet(_defaultConversionData(), 0);
     }
 
+    function testDisableInitialization() public {
+        app = new ERC20TokenStakingManager(ICMInitializable.Disallowed);
+        vm.expectRevert(abi.encodeWithSelector(Initializable.InvalidInitialization.selector));
+        app.initialize(_defaultPoSSettings(), token);
+    }
+
+    function testZeroTokenAddress() public {
+        app = new ERC20TokenStakingManager(ICMInitializable.Allowed);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                ERC20TokenStakingManager.InvalidTokenAddress.selector, address(0)
+            )
+        );
+        app.initialize(_defaultPoSSettings(), IERC20Mintable(address(0)));
+    }
+
     function testZeroMinimumDelegationFee() public {
         app = new ERC20TokenStakingManager(ICMInitializable.Allowed);
         vm.expectRevert(
             abi.encodeWithSelector(PoSValidatorManager.InvalidDelegationFee.selector, 0)
         );
-        app.initialize(
-            PoSValidatorManagerSettings({
-                baseSettings: ValidatorManagerSettings({
-                    subnetID: DEFAULT_SUBNET_ID,
-                    churnPeriodSeconds: DEFAULT_CHURN_PERIOD,
-                    maximumChurnPercentage: DEFAULT_MAXIMUM_CHURN_PERCENTAGE
-                }),
-                minimumStakeAmount: DEFAULT_MINIMUM_STAKE_AMOUNT,
-                maximumStakeAmount: DEFAULT_MAXIMUM_STAKE_AMOUNT,
-                minimumStakeDuration: DEFAULT_MINIMUM_STAKE_DURATION,
-                minimumDelegationFeeBips: 0,
-                maximumStakeMultiplier: DEFAULT_MAXIMUM_STAKE_MULTIPLIER,
-                weightToValueFactor: DEFAULT_WEIGHT_TO_VALUE_FACTOR,
-                rewardCalculator: IRewardCalculator(address(0)),
-                uptimeBlockchainID: DEFAULT_SOURCE_BLOCKCHAIN_ID
-            }),
-            token
-        );
+
+        PoSValidatorManagerSettings memory defaultPoSSettings = _defaultPoSSettings();
```

Comment:
I like this minimization of repetition and it's also way easier to see what the value/case being tested is. 👍